src/main/scala/latis/ops/ToHapiTime.scala

```diff
@@ -40,43 +40,43 @@ class ToHapiTime extends UnaryOperation {
         Time(md)
       }
 
-      tHapi.fold(
-        _ => throw new RuntimeException("Failed to apply to model"),
-        Function(_, r)
-      )
-    case _ => throw new RuntimeException("Failed to apply to model")
+      Either.catchOnly[LatisException] {
+        tHapi.fold(
+          _ => throw LatisException("Failed to apply to model"),
+          Function(_, r)
+        )
+      }
```

Comment:
I think you can get rid of `Either.catchOnly` and `throw` and just use `tHapi.fold`.

Reply:
Had to give a name to the "Right" case in the fold (couldn't use _ twice) but yep that worked.

```scala
tHapi.fold(
  _ => Left(LatisException("Failed to apply to model")),
  time => Right(Function(time, r))
)
```